app/scripts/controllers/transactions/index.js

```diff
@@ -405,7 +405,9 @@ export default class TransactionController extends EventEmitter {
    * @returns {Promise<object>} resolves with txMeta
    */
   async addTxGasDefaults(txMeta, getCodeResponse) {
-    const eip1559Compatibility = await this.getEIP1559Compatibility();
+    const eip1559Compatibility =
+      (await this.getEIP1559Compatibility()) &&
+      txMeta.txParams.type !== TRANSACTION_ENVELOPE_TYPES.LEGACY;
```

Comment:
Can we switch the order of these two conditions so that we can avoid the `await` if a simple transaction type is deemed `false`?

Reply:
Good point, I will do that.